tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/Services/GoLanguageRepoServiceTests.cs

```diff
@@ -0,0 +1,138 @@
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Azure.Sdk.Tools.Cli.Services;
+using System.Threading.Tasks; // added for async SetUp
+
+namespace Azure.Sdk.Tools.Cli.Tests.Services
+{
+    internal class GoLanguageRepoServiceTests
+    {
+        private string GoPackageDir { get; set; }
+        private string GoProgram => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "go.exe" : "go";
+
+        private GoLanguageRepoService LangService { get; set; }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            var paths = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) ?? Array.Empty<string>();
+            bool found = paths.Any(p => File.Exists(Path.Combine(p, GoProgram)));
+            if (!found)
+            {
+                Assert.Ignore("No go tooling in path, can't run Go language specific language tests");
+            }
+        }
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            GoPackageDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
+            Directory.CreateDirectory(GoPackageDir);
+
+            LangService = new GoLanguageRepoService(GoPackageDir);
+
+            var resp = await LangService.CreateEmptyPackage("untitleddotloop");
```

Comment:
`untitleddotloop` ?

Reply:
It's one of our inside jokes on the inner loop team. @l0lawrence can explain.